core/src/value/primitive.rs

```diff
@@ -1974,7 +2000,174 @@ impl PrimitiveValue {
         }
     }
 
-    /// Retrieve a single DICOM time from this value.
+    /// Retrieve a single `DicomDate` from this value.
+    ///
+    /// If the value is already represented as a date, it is converted into DicomDate.
+    /// If the value is a string or sequence of strings,
+    /// the first string is decoded to obtain a DicomDate, potentially failing if the
+    /// string does not represent a valid DicomDate.
+    /// If the value is a sequence of U8 bytes, the bytes are
+    /// first interpreted as an ASCII character string.
+    ///
+    /// Unlike Rust's `chrono::NaiveDate`, `DicomDate` allows for missing date components.
+    /// DicomDate implements `AsRange` trait, so specific `chrono::NaiveDate` values can be retrieved.
+    /// - .exact()
+    /// - .earliest()
+    /// - .latest()
+    /// - .range()
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use dicom_core::value::{C, PrimitiveValue};
+    /// # use smallvec::smallvec;
+    /// # use chrono::NaiveDate;
+    ///  use dicom_core::value::{AsRange, DicomDate};
+    ///
+    ///  let value = PrimitiveValue::Str("200002".into());
+    ///  let dicom_date = value.to_dicom_date().unwrap();
+    ///
+    ///  assert_eq!(
```

Comment:
Since this is documentation, we can guide the user within the example with comments.
```suggestion
    ///  // it is not precise, day of the month is unspecified
    ///  assert_eq!(
```